messagebird/client.py

```diff
@@ -296,5 +313,22 @@ def conversation_read_webhook(self, id):
         uri = CONVERSATION_WEB_HOOKS_PATH + '/' + str(id)
         return ConversationWebhook().load(self.request(uri, 'GET', None, CONVERSATION_TYPE))
 
+    def voice_recording_list_recordings(self, call_id, leg_id):
+        uri = VOICE_API_ROOT + '/' + VOICE_PATH + '/' + str(call_id) + '/' + VOICE_LEGS_PATH + '/' + str(leg_id) + '/' + VOICE_RECORDINGS_PATH
+        return VoiceRecordingsList().load(self.request(uri, 'GET'))
+
+    def voice_recording_view(self, call_id, leg_id, recording_id):
+        uri = VOICE_API_ROOT + '/' + VOICE_PATH + '/' + str(call_id) + '/' + VOICE_LEGS_PATH + '/' + str(leg_id) + '/' + VOICE_RECORDINGS_PATH + '/' + str(recording_id)
+        recording_response = self.request(uri, 'GET')
+        recording_response['data'][0]['_links'] = recording_response['_links']
+        return VoiceRecording().load(recording_response['data'][0])
+
+    def voice_recording_download(self, call_id, leg_id, recording_id):
+        uri = VOICE_API_ROOT + '/' + VOICE_PATH + '/' + str(call_id) + '/' + VOICE_LEGS_PATH + '/' + str(leg_id) + '/' + VOICE_RECORDINGS_PATH + '/' + str(recording_id)
+        recording_response = self.request(uri, 'GET')
+        if recording_response['_links'] is not None:
+            recording_file = self.request_store_as_file(VOICE_API_ROOT + recording_response['_links']['file'], recording_response['data'][0]['id'] + '.wav')
```

Comment:
According to https://developers.messagebird.com/api/voice-calling/#recordings
The file property is under `recording_response['data'][0]['_links']['file']`

Reply:
Yes and no: according to the API spec there should be both a _links property within the data, but also as a separate _links property within the root of the response. When I tested it I found that the property within the data was omitted, while the root property was present. That's why I actually merged this one within the voice_recording_view and with the download use the root property directly.